src/Service.Tests/SqlTests/GraphQLQueryTests/GraphQLQueryTestBase.cs

```diff
@@ -1450,6 +1458,92 @@ public virtual async Task TestInvalidFilterParamQuery()
             SqlTestHelper.TestForErrorInGraphQLResponse(result.ToString());
         }
 
+        /// <summary>
+        /// Test to check that sourceFields and targetFields for relationship provided in the config
+        /// overrides relationship fields defined in DB.
+        /// In this Test the result changes when we override the source and target fields in the config.
+        /// </summary>
+        [TestMethod]
+        public async Task TestConfigTakesPrecedenceForRelationshipFieldsOverDB(
+            string[] sourceFields,
+            string[] targetFields,
+            int club_id,
+            string club_name,
+            DatabaseType dbType,
+            string testEnvironment)
+        {
+            RuntimeConfig configuration = ConfigurationTests.InitBasicRuntimeConfigWithNoEntity(dbType, testEnvironment);
+
+            Entity clubEntity = new(
+                Source: JsonSerializer.SerializeToElement("clubs"),
+                Rest: true,
+                GraphQL: true,
+                Permissions: new PermissionSetting[] { ConfigurationTests.GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+                Relationships: null,
+                Mappings: null
+            );
+
+            configuration.Entities.Add("Club", clubEntity);
```

Comment:
Since we are adding these entities explicitly for the test, why do we need to modify the pipeline configs through config-generators/checked in config files?

Reply:
it wasn't required. I did it just for consistency. but It makes sense to keep changes minimilistic.